### PR TITLE
GH-35106: [R] Docs for 11.0.0 changelog not updated

### PR DESCRIPTION
**Pull request by thisisnic**
The Changelog for the R docs wasn't updated in the last release; I think this is due to the timing of updates, and things being merged after the release scripts were ran.

Fix apache/arrow#35106

### CODE REVIEW

File: docs/r/news/index.html [hunk 1]
```diff
@@ -25,7 +25,7 @@
     <a class="navbar-brand me-2" href="../index.html">Arrow R Package</a>
 
     <span class="version">
-      <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">11.0.0.2</small>
+      <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">11.0.0.9000</small>
```
Comment (kou) [inline docs/r/news/index.html:28]: Is this intentional?
I think that this should be `11.0.0.3`.

File: docs/r/news/index.html [hunk 2]
```diff
@@ -69,32 +69,115 @@ <h6 class="dropdown-header" data-toc-skip>Installation</h6>
...
 <div class="row">
   <main id="main" class="col-md-9"><div class="page-header">
       <img src="" class="logo" alt=""><h1>Changelog</h1>
-      <small>Source: <a href="https://github.com/apache/arrow/blob/master/r/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
+      <small>Source: <a href="https://github.com/apache/arrow/blob/main/r/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
     </div>
 
-    <div class="section level2"><h2 class="pkg-version" data-toc-text="11.0.0" id="arrow-1100">arrow 11.0.0<a class="anchor" aria-label="anchor" href="#arrow-1100"></a></h2></div>
+    <div class="section level2">
+<h2 class="pkg-version" data-toc-text="11.0.0.9000" id="arrow-11009000">arrow 11.0.0.9000<a class="anchor" aria-label="anchor" href="#arrow-11009000"></a></h2>
```
Comment (kou) [inline docs/r/news/index.html:90]: Ditto.

File: docs/r/news/index.html (hunk 1 above)
Comment (thisisnic) [inline docs/r/news/index.html:28]:
```suggestion
      <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">11.0.0.3</small>
```

File: docs/r/news/index.html (hunk 2 above)
Comment (thisisnic) [reply docs/r/news/index.html]: Done!

**Review by kou — APPROVED** [review]
+1

**Pull request merged**